### PR TITLE
Support years_between, months_between, weeks_between, and days_between APIs.

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

Added below date utility functions that calculate duration between start_date and end_date

- years_between
- months_between
- weeks_between
- days_between

# Checklist
Make sure you have performed the following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/util.R [hunk 1]
```diff
@@ -2417,6 +2417,47 @@ week <- function(date, unit="year") {
...
+  time_between(start_date, end_date, unit = "weeks")
+}
+
+#' API to calculate duration between the start_date and the end_date in days.
+days_between <- function(start_date, end_date=lubridate::today()) {
+  time_between(start_date, end_date, unit = "days")
+}
+
+#' API to calculate duration between the start_date and the end_date in hours
+hours_between <- function(start_date, end_date=lubridate::today()) {
```
Comment (hideaki) [inline R/util.R:2447]: Use now().

File: R/util.R [hunk 2]
```diff
@@ -2417,6 +2417,47 @@ week <- function(date, unit="year") {
   }
 }
 
+
+#' API to calculate duration between the start_date and the end_date in the provided time unit.
+time_between <- function(start_date, end_date=lubridate::today(), unit = "years") {
```
Comment (hideaki) [inline R/util.R:2422]: end_date default here looks unnecessary for this intermediate function.

File: R/util.R (hunk 1 above)
Comment (hidekoji) [reply R/util.R]: Changed these to use `now()` as suggested.

File: R/util.R (hunk 2 above)
Comment (hidekoji) [reply R/util.R]: Removed default for end_date argument.

**Pull request merged**